File: pythonsnippets/primeFactor.py




def primeFactor():
    number = int(raw_input("What number would you like to observe the prime factorization for? :"))
    
    print isPrime(number)
def isPrime(number):
    mult = []
    for items in range(1, number):
        if number % items == 0:
            tupl = items, number / items
            mult.append(tupl)
    if(len(mult)) <= 2:
        return str(number) + ' '
    else:
        return isPrime(mult[1][0]) + isPrime(mult[1][1])

primeFactor()

File: elixirsnippets/primeFactor.exs

def primeFactor()
    print "What number would you like to observe the prime factorization for? :"
    num = gets.chomp.to_i
    puts isPrime(num)
end

def isPrime(number)
    mult = []
    (1..number).each do |items|
        if number % items == 0
            tupl = items, number / items # TIL. I was expecting a tuple and the commas turned it into a list.
            mult << tupl
        end
    end
    if(mult.length) <= 2
        return number.to_s + ' '
    else
        return isPrime(mult[1][0]) + isPrime(mult[1][1])
    end
end

primeFactor()
